optuna/visualization/matplotlib/_parallel_coordinate.py

```diff
@@ -1,11 +1,130 @@
+from collections import defaultdict
+from typing import DefaultDict
+from typing import List
+from typing import Optional
+
+import numpy as np
+
 from optuna._experimental import experimental
+from optuna.logging import get_logger
+from optuna.study import Study
+from optuna.trial import TrialState
 from optuna.visualization.matplotlib._matplotlib_imports import _imports
 
 
 if _imports.is_successful():
     from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import LineCollection
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
 
 
 @experimental("2.2.0")
-def plot_parallel_coordinate() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_parallel_coordinate(study: Study, params: Optional[List[str]] = None) -> "Axes":
+    """Plot the high-dimentional parameter relationships in a study with Matplotlib.
+
+    .. seealso::
+        Please refer to :func:`optuna.visualization.plot_parallel_coordinate` for an example.
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        params:
+            Parameter list to visualize. The default is all parameters.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+    """
+
+    _imports.check()
+    return _get_parallel_coordinate_plot(study, params)
+
+
+def _get_parallel_coordinate_plot(study: Study, params: Optional[List[str]] = None) -> "Axes":
+
+    # Set up the graph style.
+    fig, ax = plt.subplots()
+    cmap = plt.get_cmap("Blues")
+    ax.set_title("Parallel Coordinate Plot")
+    ax.spines["top"].set_visible(False)
+    ax.spines["bottom"].set_visible(False)
+
+    # Prepare data for plotting.
+    trials = [trial for trial in study.trials if trial.state == TrialState.COMPLETE]
+
+    if len(trials) == 0:
+        _logger.warning("Your study does not have any completed trials.")
+        return ax
+
+    all_params = {p_name for t in trials for p_name in t.params.keys()}
+    if params is not None:
+        for input_p_name in params:
+            if input_p_name not in all_params:
+                raise ValueError("Parameter {} does not exist in your study.".format(input_p_name))
+        all_params = set(params)
+    sorted_params = sorted(list(all_params))
+
+    obj_org = [t.value for t in trials]
+    obj_min = min(obj_org)
+    obj_max = max(obj_org)
+    obj_w = obj_max - obj_min  # type: ignore
```

Comment:
[question] why is the type ignored?

Reply:
I put this temporalily but forgot to remove it before submitting PR, sorry.

Reply:
Corrected.

Reply:
okay, thanks.